### ISSUE TITLE
Built-in web server returns HTTP 404 when static URLs look like tile URLs

### ISSUE THREAD

**Issue by flother** [t-rex-tileserver/t-rex#150]
When using t-rex's built-in web server to serve a Mapbox GL font glyph from a static directory like `/static/fonts/{fontstack}/{range}.pbf`, t-rex always returns HTTP 404.

I have a project that uses t-rex to serve Mapbox Vector Tiles. The project directory looks something like this:

```
project
├──  config.toml
│
└── static
   ├─  index.html
   ├─  my-style.style.json
   │
   └── fonts
      └── Open Sans Regular
         ├─ 0-255.pbf
         └─ ...
```

In the t-rex config (`config.toml`) I configure the built-in web server with:

```toml
[webserver]
  [[webserver.static]]
  dir = "./static/"
  path = "/static"
```

I use `static/index.html` to serve a Mapbox GL JS map that uses `static/my-style.style.json` as the Mapbox Style Spec config. In that file, the font glyphs are referenced like this:

```json
{
  "version": 8,
  "name": "My Style",
  "metadata": {},
  "glyphs": "/static/fonts/{fontstack}/{range}.pbf",
  "sources": {},
  "layers": {}
}
```

To tie this all together, I run `t_rex serve --config config.toml` and then open `http://localhost:6767/static/index.html` in my browser. To get a font glyph for the map, my browser makes a request to, e.g., `/static/fonts/Open%20Sans%20Regular/0-255.pbf`. This file does exist within the static directory, but the URL also matches t-rex's URL format for tiles:

    /{tileset}/{x}/{y}/{z}.pbf

- `tileset` = static
- `x` = fonts
- `y` = Open Sans Regular
- `z` = 0-255

This of course doesn't match a tileset, and so t-rex returns an HTTP 404 response.

I was scratching my head for quite some time before I realised what's happening. As a workaround I've moved the fonts to a subdirectory — `/static/foo/fonts/Open%20Sans%20Regular/0-255.pbf` — but I feel that if a static file exists at a URL it should be served by t-rex.

One solution to this would be to only match on `/{tileset}/{x}/{y}/{z}.pbf` when `x`, `y`, and `z` are integers.

Thanks for taking the time to make t-rex for us all, it's a terrific project.

**Comment by pka** [comment]
Thanks for the detailed analysis and feedback!
I'll fix this behaviour in the next release. At a first glance, moving the static url match before `/{tileset}/{x}/{y}/{z}.pbf` would solve your problem and shouldn't have any negative consequences.

**Comment by pka** [comment]
Fixed in f610e46
